regex-automata/src/nfa/thompson/range_trie.rs

```diff
@@ -296,7 +296,7 @@ impl RangeTrie {
         assert!(!ranges.is_empty());
         assert!(ranges.len() <= 4);
 
-        let mut stack = mem::replace(&mut self.insert_stack, vec![]);
+        let mut stack = std::mem::take(&mut self.insert_stack);
```

Comment:
This needs to be `core::mem::take` because `regex-automata` has a conditional dependency on `std` (and even `alloc`).

Reply:
reverted